src/fmi4c.c

```diff
@@ -3990,105 +4014,100 @@ fmiHandle *fmi4c_loadFmu(const char *fmufile, const char* instanceName)
 }
 
 
-void freeIfNotNull(const char* ptr) {
-    if(ptr != NULL) {
-        free((char*)ptr);
-    }
-}
-
-void freeVoidIfNotNull(void* ptr) {
-    if(ptr != NULL) {
-        free(ptr);
-    }
```

Comment:
I added these functions because my static code analyser complained about freeing null pointers. It should not be a problem on any modern system though, so it is probably good to remove this.

Reply:
hmm thats odd, its supposed to be OK to free null ptrs

Reply:
I read somewhere that a long time ago there were some compilers with implementations of free() that would crash if given a null pointer. All modern compilers should handle it though.